tagmedia

```diff
@@ -30,127 +48,481 @@ GetOptions(
   'add-tag=s'        => \@opt_add_tag,
   'remove-tag=s'     => \@opt_remove_tag,
   'clean'            => \$opt_clean,
+  'help'             => sub { usage 0 },
 );
 
+my $image_data;			# hash ref with image related data
+my $current_tags = [];		# current list of tags
+my $old_tags = [];		# original list of tags
+my $digest_iso;			# digest calculated over iso image
+my $digest_part;		# digest calculated over partition
+
+# Note: all '*_block' variables use 0.5 kB units.
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+$image_data->{name} = shift;
+usage 1 if $image_data->{name} eq '';
+
+$image_data->{write} = $opt_digest || defined $opt_pad || $opt_check || @opt_add_tag || @opt_remove_tag || $opt_clean;
+
 if($opt_digest =~ /^md5(sum)?$/i) {
-  $digest = Digest::MD5->new;
+  $digest_iso = Digest::MD5->new;
+  $digest_part = Digest::MD5->new;
 }
 elsif($opt_digest =~ /^sha(1|224|256|384|512)(sum)?$/i) {
```

Comment:
NP: in both regexp can be used `(?:` for sum part.

Reply:
it's not critical here